From e915215122c46bbccdb520e05439e73883e7b36a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 3 Dec 2024 16:56:36 +0800
Subject: [PATCH 128/132] AOSCOS: MIPS: loongson64: use generic VDSO clock mode
 storage for constant_timer

Follow commit e1bdb22ebe53 ("mips: vdso: Use generic VDSO clock mode
storage") (v5.7-rc1) and revise vdso_clock_mode setting in constant_timer
code for loongson64 platforms.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/constant_timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/loongson64/constant_timer.c b/arch/mips/loongson64/constant_timer.c
index 08a3000dfe84..ca8b2c14602f 100644
--- a/arch/mips/loongson64/constant_timer.c
+++ b/arch/mips/loongson64/constant_timer.c
@@ -252,7 +252,7 @@ int __init init_constant_clocksource(void)
 	res = clocksource_register_hz(&clocksource_const, freq);
 
 	sched_clock_register(read_sched_clock, 64, freq);
-	clocksource_const.archdata.vdso_clock_mode = VDSO_CLOCKMODE_CONST;
+	clocksource_const.vdso_clock_mode = VDSO_CLOCKMODE_CONST;
 
 	return res;
 }
-- 
2.47.1

